src/lib/snarky_group_map/checked_map.ml

```diff
@@ -0,0 +1,68 @@
+open Snarky
+open Snark
+
+module Params = struct
+  type 'f t = {a: 'f; b: 'f}
+end
+
+module type S = sig
+  val to_group :
+    params:'f Params.t -> m:'f m -> 'f Cvar.t -> 'f Cvar.t * 'f Cvar.t
+end
+
+module Make
+    (M : Snarky.Snark_intf.Run) (Params : sig
+        val a : M.Field.t
+        val b : M.Field.t
+    end) =
+    struct
+
+  module A = Snarky.Snark.Make(Snarky.Backends.Mnt4.Default)
```

Comment:
You don't need to instantiate a particular backend here, it takes the snarky module as the functor parameter `M`